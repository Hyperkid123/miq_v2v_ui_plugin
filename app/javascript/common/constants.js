export const MAX_LENGTH_NAMES = 24;
export const MAX_LENGTH_DESCRIPTIONS = 128;

const maxLengthIs = __('Maximum length is %s characters.');
const requiredMaxLengthIs = __('Required. Maximum length is %s characters.');
const youHaveReachedMax = __('You have reached the maximum length of %s characters.');

export const validation = {
  name: {
    help: sprintf(maxLengthIs, MAX_LENGTH_NAMES),
    requiredMessage: sprintf(requiredMaxLengthIs, MAX_LENGTH_NAMES),
    maxLength: MAX_LENGTH_NAMES,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_NAMES)
  },
  description: {
    help: sprintf(maxLengthIs, MAX_LENGTH_DESCRIPTIONS),
    maxLength: MAX_LENGTH_DESCRIPTIONS,
    maxLengthWarning: sprintf(youHaveReachedMax, MAX_LENGTH_DESCRIPTIONS)
  }
};

export const OPENSTACK = 'openstack';
export const RHV = 'rhevm';

export const V2V_TARGET_PROVIDERS = [
  {
    name: __('Red Hat Virtualization'),
    id: RHV,
    type: 'ManageIQ::Providers::Redhat::InfraManager'
  },
  {
    name: __('Red Hat OpenStack Platform'),
    id: OPENSTACK,
    type: 'ManageIQ::Providers::Openstack::CloudManager'
  }
];

export const FETCH_TARGET_COMPUTE_URLS = {
  [RHV]:
    '/api/clusters?expand=resources' +
    '&attributes=ext_management_system.emstype,v_parent_datacenter,ext_management_system.name,hosts' +
    '&filter[]=ext_management_system.emstype=rhevm',
  [OPENSTACK]: '/api/cloud_tenants?expand=resources&attributes=ext_management_system.name,ext_management_system.id'
};
