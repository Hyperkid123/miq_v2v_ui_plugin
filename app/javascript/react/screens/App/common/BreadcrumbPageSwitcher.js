import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { OverlayTrigger, Popover, Button, Icon, ListGroup, ListGroupItem } from 'patternfly-react';

const BreadcrumbPageSwitcher = ({ items, activeHref }) => {
  const popover = (
    <Popover id="breadcrumb-switcher-popover">
      <ListGroup style={{ marginBottom: 0, borderTop: 0 }}>
        {items.map(({ name, href }) => (
          <ListGroupItem className={classNames('no-border', { active: href === activeHref })} href={href} key={href}>
            {name}
          </ListGroupItem>
        ))}
      </ListGroup>
    </Popover>
  );

  return (
    <OverlayTrigger rootClose trigger="click" placement="bottom" overlay={popover}>
      <Button
        style={{
          marginTop: '-2px',
          marginBottom: '-4px',
          marginLeft: '15px',
          height: '22px',
          width: '32px',
          lineHeight: '14px'
        }}
      >
        <Icon type="fa" name="exchange" />
      </Button>
    </OverlayTrigger>
  );
};

BreadcrumbPageSwitcher.propTypes = {
  activeHref: PropTypes.string.isRequired,
  items: PropTypes.arrayOf(PropTypes.shape({ name: PropTypes.string, href: PropTypes.string })).isRequired
};

export default BreadcrumbPageSwitcher;
