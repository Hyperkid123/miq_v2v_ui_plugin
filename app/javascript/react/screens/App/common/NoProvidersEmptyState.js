import React from 'react';
import PropTypes from 'prop-types';
import ShowWizardEmptyState from './ShowWizardEmptyState/ShowWizardEmptyState';

const NoProvidersEmptyState = ({ className }) => (
  <ShowWizardEmptyState
    className={className}
    description={
      __('Before attempting a migration, you must have at least one VMware and one Red Hat Virtualization or Red Hat OpenStack Platform provider configured.') // prettier-ignore
    }
    action={
      <React.Fragment>
        <a href="/ems_infra/show_list">{__('View VMware and Red Hat Virtualization providers')}</a>
        <br />
        <a href="/ems_cloud/show_list">{__('View Red Hat OpenStack Platform providers')}</a>
      </React.Fragment>
    }
  />
);

NoProvidersEmptyState.propTypes = {
  className: PropTypes.string
};

export default NoProvidersEmptyState;
