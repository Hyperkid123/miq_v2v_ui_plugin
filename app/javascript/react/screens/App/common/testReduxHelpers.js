import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { createStore, applyMiddleware, combineReducers } from 'redux';

const middlewares = [thunk, promiseMiddleware()];

export const generateStore = (reducers, initialState) =>
  createStore(combineReducers(reducers), initialState, applyMiddleware(...middlewares));
