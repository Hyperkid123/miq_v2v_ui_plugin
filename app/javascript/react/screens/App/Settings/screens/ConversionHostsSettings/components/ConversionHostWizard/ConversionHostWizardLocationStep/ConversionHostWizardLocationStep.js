import React from 'react';
import PropTypes from 'prop-types';
import { reduxForm, Field, reset } from 'redux-form';
import { required } from 'redux-form-validators';
import { Form } from 'patternfly-react';

import { BootstrapSelect } from '../../../../../../common/forms/BootstrapSelect';
import { V2V_TARGET_PROVIDERS, RHV, OPENSTACK } from '../../../../../../../../../common/constants';

import { stepIDs, PROVIDER_TYPES } from '../ConversionHostWizardConstants';

export const ConversionHostWizardLocationStep = ({ locationStepForm }) => {
  const selectedProviderType = locationStepForm.values && locationStepForm.values.providerType;
  const selectedProvider = locationStepForm.values && locationStepForm.values.provider;

  const providers = []; // TODO /api/providers filtered by PROVIDER_TYPES?
  const rhvClusters = []; // TODO
  const ospProjects = []; // TODO

  const selectFieldBaseProps = {
    component: BootstrapSelect,
    labelWidth: 2,
    controlWidth: 9,
    inline_label: true,
    required: true
  };

  return (
    <Form className="form-horizontal">
      <Field
        {...selectFieldBaseProps}
        name="providerType"
        label={__('Provider Type')}
        options={V2V_TARGET_PROVIDERS}
        option_key="id"
        option_value="name"
      />
      {selectedProviderType && (
        <Field
          {...selectFieldBaseProps}
          name="provider"
          label={__('Provider')}
          options={providers}
          option_key="id"
          option_value="name"
        />
      )}
      {selectedProviderType === RHV && (
        <Field
          {...selectFieldBaseProps}
          name="cluster"
          label={__('Cluster')}
          options={rhvClusters}
          option_key="id"
          option_value="name"
          disabled={!selectedProvider}
        />
      )}
      {selectedProviderType === OPENSTACK && (
        <Field
          {...selectFieldBaseProps}
          name="project"
          label={__('Project')}
          options={ospProjects}
          option_key="id"
          option_value="name"
          disabled={!selectedProvider}
        />
      )}
    </Form>
  );
};

ConversionHostWizardLocationStep.propTypes = {
  locationStepForm: PropTypes.object
};

ConversionHostWizardLocationStep.defaultProps = {
  locationStepForm: {}
};

export default reduxForm({
  form: stepIDs.locationStep,
  destroyOnUnmount: false
})(ConversionHostWizardLocationStep);
