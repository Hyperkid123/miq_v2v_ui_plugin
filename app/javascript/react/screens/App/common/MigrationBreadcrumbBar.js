import React from 'react';
import PropTypes from 'prop-types';
import { Breadcrumb } from 'patternfly-react';
import Toolbar from '../../../config/Toolbar';
import BreadcrumbPageSwitcher from './BreadcrumbPageSwitcher';

const items = [
  { name: __('Migration Plans'), href: '#/plans' },
  { name: __('Infrastructure Mappings'), href: '#/mappings' },
  { name: __('Migration Settings'), href: '#/settings' }
];

const MigrationBreadcrumbBar = ({ activeHref }) => {
  const activeItem = items.find(item => item.href === activeHref);
  return (
    <Toolbar>
      <Breadcrumb.Item active>{__('Migration')}</Breadcrumb.Item>
      <Breadcrumb.Item active>
        <strong>{activeItem.name}</strong>
      </Breadcrumb.Item>
      <BreadcrumbPageSwitcher items={items} activeHref={activeHref} />
    </Toolbar>
  );
};

MigrationBreadcrumbBar.propTypes = {
  activeHref: PropTypes.string.isRequired
};

export default MigrationBreadcrumbBar;
