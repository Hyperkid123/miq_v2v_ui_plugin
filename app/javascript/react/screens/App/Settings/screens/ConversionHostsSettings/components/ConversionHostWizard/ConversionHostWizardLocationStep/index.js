import { connect } from 'react-redux';
import ConversionHostWizardLocationStep from './ConversionHostWizardLocationStep';

import * as SettingsActions from '../../../../../SettingsActions';

import { stepIDs } from '../ConversionHostWizardConstants';

const mapStateToProps = ({
  form,
  settings: { isFetchingProviders, providers, isFetchingTargetComputeResources, targetComputeResources }
}) => ({
  locationStepForm: form[stepIDs.locationStep],
  isFetchingProviders,
  providers,
  isFetchingTargetComputeResources,
  targetComputeResources
});

const mergeProps = (stateProps, dispatchProps, ownProps) => Object.assign(stateProps, ownProps.data, dispatchProps);

export default connect(
  mapStateToProps,
  SettingsActions,
  mergeProps
)(ConversionHostWizardLocationStep);
