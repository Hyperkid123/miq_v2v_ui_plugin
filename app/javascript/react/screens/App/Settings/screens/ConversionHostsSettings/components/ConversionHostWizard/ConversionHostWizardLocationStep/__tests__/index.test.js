import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { reducer as formReducer } from 'redux-form';

import { generateStore } from '../../../../../../../common/testReduxHelpers';
import ConversionHostWizardLocationStep from '../ConversionHostWizardLocationStep';
import ConversionHostWizardLocationStepContainer from '../index';
import { RHV } from '../../../../../../../../../../common/constants';

describe('ConversionHostWizardLocationStep integration test', () => {
  const store = generateStore(
    { form: formReducer },
    {
      form: { conversionHostWizardLocationStep: { values: { providerType: RHV, provider: '1' } } },
      providers: { providers: [{ id: '1', name: 'Mock Provider', type: 'ManageIQ::Providers::Redhat::InfraManager' }] },
      targetResources: { targetClusters: [], isFetchingTargetClusters: false }
    }
  );

  const mountComponent = () =>
    mount(
      <Provider store={store}>
        <ConversionHostWizardLocationStepContainer />
      </Provider>
    );

  it('should mount ConversionHostWizardLocationStep with mapStateToProps reduced', () => {
    const wrapper = mountComponent();

    // query the unconnected component to assert reduced props
    const component = wrapper.find(ConversionHostWizardLocationStep);
    expect(component.props()).toMatchSnapshot();
  });
});
