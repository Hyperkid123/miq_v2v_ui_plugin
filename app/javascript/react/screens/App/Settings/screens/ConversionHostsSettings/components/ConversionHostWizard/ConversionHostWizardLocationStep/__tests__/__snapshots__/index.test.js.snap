// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversionHostWizardLocationStep integration test should mount ConversionHostWizardLocationStep with mapStateToProps reduced 1`] = `
Object {
  "fetchTargetClustersAction": [Function],
  "isFetchingTargetClusters": false,
  "providers": Array [
    Object {
      "id": "1",
      "name": "Mock Provider",
      "type": "ManageIQ::Providers::Redhat::InfraManager",
    },
  ],
  "resetFormAction": [Function],
  "selectedProviderId": "1",
  "selectedProviderType": "rhevm",
  "targetClusters": Array [],
}
`;
